### PR TITLE
add Test and OpenDirectly wrapper

### PR DESCRIPTION
**Pull request by BrendanChou**

### CODE REVIEW

File: contracts/exchange-wrappers/TestExchangeWrapper.sol [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+ */
+contract TestExchangeWrapper is
+    ExchangeWrapper
+{
+    using TokenInteract for address;
+
+    // ============ Constants ============
+
+    // arbitrary address to send the tokens to (they are burned)
+    address constant EXCHANGE_ADDRESS = address(0x1);
```
Comment (AntonioJuliano) [inline contracts/exchange-wrappers/TestExchangeWrapper.sol]: make this `public`

File: contracts/exchange-wrappers/TestExchangeWrapper.sol [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        uint256 requestedFillAmount,
+        bytes calldata orderData
+    )
+        external
+        returns (uint256)
+    {
+        Order memory order = parseData(orderData);
+
+        require(
+            order.originator == tradeOriginator,
```
Comment (AntonioJuliano) [inline contracts/exchange-wrappers/TestExchangeWrapper.sol:67]: Why are you passing all this info in in the order if it's already given in `exchange`? Just so you can validate everything else actually sends through the bytes correctly?

File: contracts/exchange-wrappers/TestExchangeWrapper.sol [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        assembly {
+            originator := mload(add(data, 32))
+            makerToken := mload(add(data, 64))
+            takerToken := mload(add(data, 92))
+            makerAmount := mload(add(data, 128))
+            takerAmount := mload(add(data, 160))
+        }
+
+        return Order({
+            originator: address(originator),
```
Comment (AntonioJuliano) [inline contracts/exchange-wrappers/TestExchangeWrapper.sol]: why are you declaring these as `uint256`'s and then casting them? Wouldn't it work if you just declared them as addresses?

File: src/exchange_wrappers/TestExchangeWrapper.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import web3Utils from 'web3-utils';
+import BN from 'bn.js';
+import BigNumber from 'bignumber.js';
+import { TestOrder } from '../types';
+import { TestExchangeWrapper as Contract } from '../../migrations/deployed.json';
+
+export class TestExchangeWrapper {
```
Comment (AntonioJuliano) [inline src/exchange_wrappers/TestExchangeWrapper.ts:7]: You need to actually instances of these to `OrderMapper.ts`, and copy everything else that's done there

**Review by AntonioJuliano — CHANGES_REQUESTED** [review]

File: contracts/exchange-wrappers/TestExchangeWrapper.sol (hunk 2 above)
Comment (BrendanChou) [reply contracts/exchange-wrappers/TestExchangeWrapper.sol:67]: yeah to help validate the fields passed in by the caller of the exchangeWrapper

**Review by AntonioJuliano — APPROVED** [review]

**Pull request merged**